soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -12699,7 +12704,13 @@ s32 func_8084DFF4(GlobalContext* globalCtx, Player* this) {
             else {
                 temp1 = temp2 = (this->getItemId == GI_HEART_PIECE) ? NA_BGM_SMALL_ITEM_GET : NA_BGM_ITEM_GET | 0x900;
             }
-            Audio_PlayFanfare(temp1);
+          
+            RandoFanfareSkip:
+            if (gSaveContext.n64ddFlag) {
+                Audio_PlayFanfare_Rando(this->getItemId);
+            } else {
+                Audio_PlayFanfare(temp1);
+            }
```

Comment:
Nitpick. You could leave the normal `Audio_PlayFanfare(temp1);` above `RandoFanFareSkip:` and only keep the if statement for the rando fanfare.

Reply:
but then it would play both the normal and the rando fanfare

Reply:
oh wait, gotos? oof

Reply:
There's this higher up in the code:

```// In rando the fanfares are handled in z_en_si.c, so we can skip down a bit
        if (gSaveContext.n64ddFlag) {
            goto RandoFanfareSkip;
        }```

Reply:
this would be better if it was just:
```
if rando and rando fanfares {
 play rando fanfare
} else if {
 rupee/heart stuff
} else {
 other item stuff
}
```

Reply:
Sounds good!  I will work on this and test again tonight.

Reply:
> There's this higher up in the code

yeah, noticed that, hence my "gotos oof" comment

Reply:
Yeah we posted it at the same time. The conditionals are ready happening in the new rando function though. Shouldn't wrapping the z_player.c stuff in a single if/else be enough instead of a goto then?